dev/Scroller/Scroller.cpp

```diff
@@ -6827,22 +7035,39 @@ winrt::InteractionTrackerInertiaRestingValue Scroller::GetInertiaRestingValue(
     winrt::hstring const& target,
     winrt::hstring const& scale) const
 {
+    bool isInertiaFromImpulse = IsInertiaFromImpulse();
     winrt::InteractionTrackerInertiaRestingValue modifier = winrt::InteractionTrackerInertiaRestingValue::Create(compositor);
-    winrt::ExpressionAnimation conditionExpressionAnimation = snapPointWrapper->CreateConditionalExpression(compositor, target, scale);
-    winrt::ExpressionAnimation restingPointExpressionAnimation = snapPointWrapper->CreateRestingPointExpression(compositor, target, scale);
+    winrt::ExpressionAnimation conditionExpressionAnimation = snapPointWrapper->CreateConditionalExpression(m_interactionTracker, target, scale, isInertiaFromImpulse);
+    winrt::ExpressionAnimation restingPointExpressionAnimation = snapPointWrapper->CreateRestingPointExpression(m_interactionTracker, target, scale, isInertiaFromImpulse);
 
     modifier.Condition(conditionExpressionAnimation);
     modifier.RestingValue(restingPointExpressionAnimation);
 
     return modifier;
 }
 
-bool Scroller::IsLoaded()
+// Relies on InteractionTracker.IsInertiaFromImpulse starting with RS5,
+// returns the replacement field m_isInertiaFromImpulse otherwise.
+bool Scroller::IsInertiaFromImpulse() const
+{
+    if (m_interactionTracker)
+    {
+        winrt::IInteractionTracker4 interactionTracker4 = m_interactionTracker;
+
+        return interactionTracker4 ? interactionTracker4.IsInertiaFromImpulse() : m_isInertiaFromImpulse;
```

Comment:
The result of the instantiated interactionTracker4 will be null if m_interactionTracker is null, or if it is not on a version where the interface is present so we can move the declaration to the if check.

if(winrt::IInteractionTracker4 interactionTracker4 = m_interactionTracker)
{
    return interactionTracker4.IsInertiaFromImpulse();
}